### PR TITLE
Added support for the IBM 6746 Typewriter

### PR DESCRIPTION
**Pull request by Justin-sandwich**
## Description

Added support for the IBM 6746 typewriter, based on similar non-QMK projects pertaining to IBM wheelwriters and similar typewriter/ribbon cable - to - microcontroller projects.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ibm_6746/ibm_6746.c
Comment (fauxpark) [inline keyboards/ibm_6746/ibm_6746.c:1]: This file can be deleted.

File: keyboards/ibm_6746/config.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+//#define MATRIX_ROW_PINS { GP12, GP13, GP14, GP15, GP16, GP21, GP23, GP20 }
+//                  COLS: 1    2    3    4    5    6    7    8    9     10    11    12    13
+//#define MATRIX_COL_PINS { GP9, GP8, GP7, GP6, GP5, GP4, GP3, GP2, GP22, GP26, GP27, GP28, GP29 }
+//#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+//#define DIODE_DIRECTION COL2ROW
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed (5 is default) */
+#define DEBOUNCE 5
```
Comment (fauxpark) [inline keyboards/ibm_6746/config.h:32]:
```suggestion
```

File: keyboards/ibm_6746/info.json
Comment (fauxpark) [inline keyboards/ibm_6746/info.json:1]: Please run `qmk format-json -i` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ibm_6746/readme.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# ibm_6746
+
+![ibm_6746](https://i.imgur.com/mrXTFi9.png)
+![circuit board:](https://i.imgur.com/QPzkDhs.png)
```
Comment (zvecr) [inline keyboards/ibm_6746/readme.md:4]: This project should be relocated to the handwired folder

File: keyboards/ibm_6746/rules.mk [hunk 3]
```diff
@@ -0,0 +1,2 @@
+
+# this file is intentionally left blank
```
Comment (zvecr) [inline keyboards/ibm_6746/rules.mk:2]:
```suggestion
# this file is intentionally left blank
```

File: keyboards/ibm_6746/ibm_6746.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+//  Mar_Rel  +-        1 2 3 4 5 6 7 8 9 0 - #  Backspace  Paper_Up   
+//  Lmar     Tab       Q W E R T Y U I O P ½ [  CRtn       Paper_Down 
+//  Rmar     Lock      A S D F G H J K L ; * ⅔             Micro_Up
+//  T_Set    LShift >< Z X C V B N M , . /?      RShift    Micro_Down
+//  T_Clr              Code  Space               <X        Line_Space (dead)
+
+//  K3N (right shift) is skipped as it is electronically identical to K3B (left shift).
+//  K4P (line space) is skipped as it is not found on the keyboard electronically.
+
+#define LAYOUT(	\
```
Comment (zvecr) [inline keyboards/ibm_6746/ibm_6746.h:31]: Layout macros must be configured within info.json, and this file removed.

File: keyboards/ibm_6746/config.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Justin (@Justin-sandwich)
+// SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (zvecr) [inline keyboards/ibm_6746/config.h:2]:
```suggestion
// Copyright 2023 Justin (@Justin-sandwich)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/ibm_6746/config.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+#pragma once
+
+/* key matrix size */
+#define MATRIX_ROWS 8
+#define MATRIX_COLS 13
+
```
Comment (zvecr) [inline keyboards/ibm_6746/config.h:9]:
```suggestion
```

File: keyboards/ibm_6746/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,18 @@
+#include QMK_KEYBOARD_H
```
Comment (zvecr) [inline keyboards/ibm_6746/keymaps/default/keymap.c:1]:
```suggestion
// Copyright 2023 Justin (@Justin-sandwich)
// SPDX-License-Identifier: GPL-2.0-or-later
#include QMK_KEYBOARD_H
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/ibm_6746/config.h [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
```
Comment (drashna) [inline keyboards/ibm_6746/config.h:23]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ibm_6746/ibm_6746.h (hunk 4 above)
Comment (drashna) [reply keyboards/ibm_6746/ibm_6746.h:31]: `qmk info -f json` can help with that

**Pull request closed without merging**